src/main/java/org/assertj/core/internal/ObjectArrays.java

```diff
@@ -669,6 +673,27 @@ public void assertDoesNotContainNull(AssertionInfo info, Object[] actual) {
     arrays.assertHasOnlyElementsOfTypes(info, failures, actual, types);
   }
 
+  public <E> void assertHasExactlyElementsOfTypes(AssertionInfo info, E[] actual, Class<?>... types) {
+    Objects.instance().assertNotNull(info, actual);
+    List<Object> actualTypeList = extractTypeList(asList(actual));
+    IterableDiff diff = diff(actualTypeList, asList(types), getComparisonStrategy());
```

Comment:
`getComparisonStrategy()` would return the comparison strategy to compare elements not their class.
The tests work because StandardComparisonStrategy is used by default.

Here we should pass `StandardComparisonStrategy.instance()` and put a comment explaining why (because we don't compare elements but their types)

Reply:
Add test with a custom comparison strategy to make sure this does not break the assertion